### PR TITLE
SONARJAVA-2776 FP on S4449 with latest versions of Guava (fixes SONARJAVA-2777 and SONARJAVA-2866 Semantic: Support annotation applicable on ElementType.TYPE_USE)

### PR DESCRIPTION
**Pull request by alban-auzeill**

### CODE REVIEW

File: java-frontend/src/test/java/org/sonar/java/resolve/targets/TypeAnnotationOnMembers.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ */
+package org.sonar.java.resolve.targets;
+
+import java.util.List;
+
+public class TypeAnnotationOnMembers<C> {
+
+  // Note: bellow annotations @TypeAnnotation("t") are not present in our semantic
+  // BytecodeFieldVisitor.visitTypeAnnotation does not support typePath != null
+  // BytecodeMethodVisitor.visitTypeAnnotation does not support the METHOD_TYPE_PARAMETER case
```
Comment (benzonico) [inline java-frontend/src/test/java/org/sonar/java/resolve/targets/TypeAnnotationOnMembers.java]: I struggle to understand what this comment means.

Comment (benzonico) [reply java-frontend/src/test/java/org/sonar/java/resolve/targets/TypeAnnotationOnMembers.java]: And from what I can understand from it, it seems to me that the support of type annotation you fixed was not complete ?

File: java-frontend/src/test/files/se/ParameterNullnessCheck.java [hunk 2]
```diff
@@ -33,10 +33,14 @@ void foo(Object o) { // flow@foo [[order=2]] {{Method 'foo' declaration.}}
       null); // flow@A [[order=1]] {{Argument can be null.}}
     B b = new B();
 
-    Preconditions.checkNotNull( // Noncompliant [[sc=19;ec=31;flows=checkNotNull]] 
+    Preconditions.checkNotNull( // Noncompliant [[sc=19;ec=31;flows=checkNotNull]]
       null); // flow@checkNotNull [[order=1]] {{Argument can be null.}}
   }
 
+  boolean checkerFrameworkNullableCompatibility(@Nullable String javaxNullable) {
```
[the remark this answers is not in the document]
Comment (benzonico) [reply java-frontend/src/test/files/se/ParameterNullnessCheck.java]: ok. in fact I do understand but I would rather see an explicit testcase in addition to this indirect one that would still deserve a comment.

File: java-frontend/src/test/java/org/sonar/java/resolve/targets/TypeAnnotationOnMembers.java (hunk 1 above)
Comment (alban-auzeill) [reply java-frontend/src/test/java/org/sonar/java/resolve/targets/TypeAnnotationOnMembers.java]: I changed and moved the comment to unit tests and visitors code to ease the understanding by being close to the context.

File: java-frontend/src/test/files/se/ParameterNullnessCheck.java [hunk 3]
```diff
@@ -33,10 +33,14 @@ void foo(Object o) { // flow@foo [[order=2]] {{Method 'foo' declaration.}}
       null); // flow@A [[order=1]] {{Argument can be null.}}
     B b = new B();
 
-    Preconditions.checkNotNull( // Noncompliant [[sc=19;ec=31;flows=checkNotNull]] 
+    Preconditions.checkNotNull( // Noncompliant [[sc=19;ec=31;flows=checkNotNull]]
```
[the remark this answers is not in the document]
Comment (alban-auzeill) [reply java-frontend/src/test/files/se/ParameterNullnessCheck.java]: removed

File: java-frontend/src/test/files/se/ParameterNullnessCheck.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (alban-auzeill) [reply java-frontend/src/test/files/se/ParameterNullnessCheck.java]: I added an explicit testcase

**Pull request merged**